src/utils/DepositUtils.ts

```diff
@@ -116,17 +99,24 @@ export async function queryHistoricalDepositForFill(
   if (isDefined(cachedDeposit)) {
     deposit = cachedDeposit as DepositWithBlock;
   } else {
-    deposit = await spokePoolClient.findDeposit(fill.depositId, fill.destinationChainId);
+    const result = await spokePoolClient.findDeposit(fill.depositId);
+    if (!result.found) {
+      return result;
+    }
+
+    ({ deposit } = result);
     if (cache) {
-      await setDepositInCache(deposit, getCurrentTime(), cache, DEFAULT_CACHING_TTL);
+      await setDepositInCache(deposit!, getCurrentTime(), cache, DEFAULT_CACHING_TTL);
```

Comment:
Nasty - fix this.

Reply:
https://github.com/across-protocol/sdk/pull/910/commits/2adc1aab97c2d5d1e7dff761c8b7ce245576983a